### DISCUSSION TITLE
Transform_Func Error for Plane Stress Study

### DISCUSSION THREAD

**Discussion by racheljhur** [idaholab/moose#27293] [Q&A Modules: Solid mechanics]
Hi everyone, 

I am performing a plane stress simulation with periodic boundary conditions (for an SVE) and an applied uniaxial load equivalent to 1 MPa. Before I add more complex microstructure geometry, I am trying to get a simple baseline code to work on a 2x2 square. I followed the example on MooseDocs for weak_plane_stress, but adding PBCs and an applied traction as a BC is prompting this error: 

*** ERROR ***
The parameter "transform_func" is being retrieved before being set.

This is my script for reference:
[GlobalParams]
  displacements = 'disp_x disp_y'
  out_of_plane_strain = strain_zz
[]

[Mesh]
  [./square]
    type = GeneratedMeshGenerator
    dim = 2
    nx = 2
    ny = 2
  [../]
[]

[Variables]
  [./disp_x]
  [../]
  [./disp_y]
  [../]
  [./strain_zz]
    order = FIRST
    family = SCALAR
  [../]
[]

[AuxVariables]
  [./nl_strain_zz]
    order = CONSTANT
    family = MONOMIAL
  [../]
[]

[Postprocessors]
  [./react_z]
    type = MaterialTensorIntegral
    rank_two_tensor = stress
    index_i = 2
    index_j = 2
  [../]
  [./min_strain_zz]
    type = NodalExtremeValue
    variable = strain_zz
    value_type = min
  [../]
  [./max_strain_zz]
    type = NodalExtremeValue
    variable = strain_zz
    value_type = max
  [../]
[]

[Physics/SolidMechanics/QuasiStatic]
  [plane_stress]
    planar_formulation = WEAK_PLANE_STRESS
    strain = FINITE
    generate_output = 'stress_xx stress_xy stress_yy stress_zz strain_xx strain_xy strain_yy'
  []
[]

[AuxKernels]
  [./strain_zz]
    type = RankTwoAux
    rank_two_tensor = total_strain
    variable = nl_strain_zz
    index_i = 2
    index_j = 2
  [../]
[]

[BCs]
  [Periodic]
    [x]
      variable = disp_x
      primary = 'right'
      secondary = 'left'
      translation = '-2 0 0'
    []
    [y]
      variable = disp_y
      primary = 'top'
      secondary = 'bottom'
      translation = '0 -2 0'
    []
    [uniaxial_traction_x]
      type = NeumannBC
      variable = disp_x
      boundary = right
      value = 1e6
    []
    # constrain rigid body motion
    [fix_all_x]
      type = DirichletBC
      boundary = 'right'
      variable = disp_x
      value = 0
    []
    [fix_all_y]
      type = DirichletBC
      boundary = 'bottom'
      variable = disp_y
      value = 0
    []
  []
[]

[Materials]
  [./elastic_tensor]
    type = ComputeIsotropicElasticityTensor
    youngs_modulus = 1e9
    poissons_ratio = 0.3
  [../]

  [./stress]
    type = ComputeFiniteStrainElasticStress
  [../]
[]

[Executioner]
  type = Steady
  petsc_options_iname = '-pc_type'
  petsc_options_value = 'lu'
[]

[Outputs]
  exodus = true
[]

Any help would be much appreciated. Thank you.

**Comment by GiudGiud** [comment]
Hello

seems you need to set this transform_func parameter to the object that is returning this error
This error indicates the parameters have not been created properly in that object. We’ll have to take a look if this is an object in moose

Guillaume

**Reply by racheljhur** [reply]
I actually ended up using lagrangian kernels to homogenize as I was working with an SVE. I also canceled out appropriate terms to make it plane stress. Thank you for the response.
